Sources/ConcordiumSwiftSDK/Model/Account.swift

```diff
@@ -16,9 +49,373 @@ struct AccountAddress {
+
+    static func fromGrpcType(_ grpc: Concordium_V2_OpenStatus) -> OpenStatus {
+        OpenStatus(rawValue: grpc.rawValue)!
+    }
+}
+
+struct CommissionRates {
+    let finalization: AmountFraction
+    let baking: AmountFraction
+    let transaction: AmountFraction
+
+    static func fromGrpcType(_ grpc: Concordium_V2_CommissionRates) -> CommissionRates {
+        CommissionRates(
+                finalization: .fromGrpcType(grpc.finalization),
+                baking: .fromGrpcType(grpc.baking),
+                transaction: .fromGrpcType(grpc.transaction)
+        )
+    }
+}
+
+struct BakerPoolInfo {
+    let openStatus: OpenStatus
+    let metadataUrl: String
+    let commissionRates: CommissionRates
+
+    static func fromGrpcType(_ grpc: Concordium_V2_BakerPoolInfo) -> BakerPoolInfo {
+        BakerPoolInfo(
+                openStatus: .fromGrpcType(grpc.openStatus),
+                metadataUrl: grpc.url,
+                commissionRates: .fromGrpcType(grpc.commissionRates)
+        )
+    }
+}
+
+enum DelegationTarget {
+    case passive
+    case baker(BakerId)
+
+    static func fromGrpcType(_ grpc: Concordium_V2_DelegationTarget) -> DelegationTarget? {
+        switch grpc.target {
+        case nil:
+            return nil
+        case .passive:
+            return .passive
+        case let .baker(b):
+            return .baker(b.value)
+        }
+    }
+}
+
+enum AccountStakingInfo {
+    case baker(
+            stakedAmount: Amount,
+            restakeEarnings: Bool,
+            bakerInfo: BakerInfo,
+            pendingChange: StakePendingChange?,
+            poolInfo: BakerPoolInfo?
+    )
+    case delegated(
+            stakedAmount: Amount,
+            restakeEarnings: Bool,
+            delegationTarget: DelegationTarget,
+            pendingChange: StakePendingChange?
+    )
+
+    static func fromGrpcType(_ grpc: Concordium_V2_AccountStakingInfo) throws -> AccountStakingInfo? {
+        switch grpc.stakingInfo {
+        case nil:
+            return nil
+        case let .baker(b):
+            return .baker(
+                    stakedAmount: b.stakedAmount.value,
+                    restakeEarnings: b.restakeEarnings,
+                    bakerInfo: .fromGrpcType(b.bakerInfo),
+                    pendingChange: .fromGrpcType(b.pendingChange),
+                    poolInfo: b.hasPoolInfo ? .fromGrpcType(b.poolInfo) : nil
+            )
+        case let .delegator(d):
+            return .delegated(
+                    stakedAmount: d.stakedAmount.value,
+                    restakeEarnings: d.restakeEarnings,
+                    delegationTarget: try .fromGrpcType(d.target) ?? {
+                        throw GrpcError.requiredValueMissing("delegation target")
+                    }(),
+                    pendingChange: .fromGrpcType(d.pendingChange)
+            )
+        }
+    }
+}
+
+struct AccountInfo {
+    let accountNonce: SequenceNumber
+    let accountAmount: Amount
```

Comment:
Agree it's weird. Did it because the SDKs do (some of them at least). Should we rename?

Reply:
Well, since the context (the type) is right there, we probably won't confuse anyone by removing the "account" prefix - but it's not something I feel strongly about, so I'll leave it up to you :)

Reply:
Agree I'll do that and similar cases in other types.